dot4gravity/src/tests.rs

```diff
@@ -313,12 +324,61 @@ fn a_stone_dropped_on_a_stone() {
     ];
 
     state.board.cells = cells;
+    state.phase = GamePhase::Play;
 
     let state = Game::drop_stone(state, ALICE, Side::West, 0).unwrap();
     assert_eq!(
         state.board.get_cell(&Coordinates { row: 0, col: 0 }),
         Cell::Stone(alice_index)
     );
+    assert_eq!(
+        state.board.get_cell(&Coordinates { row: 0, col: 1 }),
+        Cell::Stone(bob_index)
+    );
+}
+
+#[test]
+fn a_stone_cannot_be_dropped_at_bounds() {
+    let mut state = Game::new_game(ALICE, BOB, Some(INITIAL_SEED));
+    let o = Cell::Empty;
+    let x = Cell::Stone(state.player_index(&BOB));
+    state.phase = GamePhase::Play;
+    state.board.cells = [
+        [x, x, x, x, x, x, x, x, x, x],
+        [x, o, o, o, o, o, o, o, o, x],
+        [x, o, o, o, o, o, o, o, o, x],
+        [x, o, o, o, o, o, o, o, o, x],
+        [x, o, o, o, o, o, o, o, o, x],
+        [x, o, o, o, o, o, o, o, o, x],
+        [x, o, o, o, o, o, o, o, o, x],
+        [x, o, o, o, o, o, o, o, o, x],
+        [x, o, o, o, o, o, o, o, o, x],
+        [x, x, x, x, x, x, x, x, x, x],
+    ];
```

Comment:
should it be tested for blocks as well? (changing the stones in the bounds to blocks)

Reply:
Yep sounds good :+1: